packages/twenty-front/src/modules/workflow/hooks/useCreateStep.tsx

```diff
@@ -43,15 +48,30 @@ export const useCreateStep = ({
       throw new Error("Can't add a node when there is no current version.");
     }
 
-    return updateOneWorkflowVersion({
-      idToUpdate: currentVersion.id,
-      updateOneRecordInput: {
-        steps: insertStep({
-          steps: currentVersion.steps ?? [],
-          parentStepId: parentNodeId,
-          stepToAdd: nodeToAdd,
-        }),
-      },
+    const updatedSteps = insertStep({
+      steps: currentVersion.steps ?? [],
+      parentStepId: parentNodeId,
+      stepToAdd: nodeToAdd,
+    });
+
+    if (workflow.currentVersion.status === 'DRAFT') {
+      await updateOneWorkflowVersion({
+        idToUpdate: currentVersion.id,
+        updateOneRecordInput: {
+          steps: updatedSteps,
+        },
+      });
+
+      return;
+    }
+
+    await createNewWorkflowVersion({
+      name: workflow.currentVersion.name.startsWith('v')
```

Comment:
this name should rely on a version count rather than this kind of parsing

Reply:
You should not store the "v". As discussed, you will have the count of versions for the workflow thanks to `totalCount`. The name you send should simply be `"v" + ${totalCount + 1}`

Reply:
Um, so I shouldn't add the `"v"` prefix here but `String(totalCount + 1)`, right? And in the UI, I'll format the version accordingly, adding a "v."

I wanted to make sure we are okay with the idea of storing an integer in a string field.